saos-persistence/src/main/java/pl/edu/icm/saos/persistence/model/Judgment.java

```diff
@@ -598,9 +605,19 @@ public void setJudgmentType(JudgmentType judgmentType) {
         this.judgmentType = judgmentType;
     }
 
-    public void setTextContent(String textContent) {
+    public void setTextContent(JudgmentTextContent textContent) {
+        if (textContent != null) {
+            textContent.setJudgment(this);
+        }
         this.textContent = textContent;
     }
+    
+    public void setRawTextContent(String rawTextContent) {
```

Comment:
This is not a good idea to have a setter like this. First it is a setter for a transient getter which is a bit misleading. Second, you change another property here implicitly.